src/UI/state/speciesInformation/actions/upsertSpeciesInfo.action.ts

```diff
@@ -0,0 +1,38 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { fetchGraphQlDataAuthenticated } from '../../../api/api';
+import {
+  speciesInformationById,
+  upsertSpeciesInformationMutation,
+} from '../../../api/queries';
+import { SpeciesInformation } from '../../state.types';
+import { AppState } from '../../store';
+import { setCurrentInfoForEditing } from '../speciesInformationSlice';
+
+export const upsertSpeciesInformation = createAsyncThunk(
+  'speciesInformation/upsert',
+  async (speciesInformation: SpeciesInformation, { getState }) => {
+    try {
+      const token = (getState() as AppState).auth.token;
+      await fetchGraphQlDataAuthenticated(
+        upsertSpeciesInformationMutation(speciesInformation),
+        token
+      );
+    } catch (e) {
+      console.log(e);
```

Comment:
Toast?

Reply:
Yes, I'll extend this in the next PR to include notications for success as well.